internal/services/voiceservices/voice_services_communications_gateway_resource_test.go

```diff
@@ -0,0 +1,264 @@
+  name     = "acctest-rg-%d"
+  location = "%s"
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r VoiceServicesCommunicationsGatewayTestResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_voice_services_communications_gateway" "test" {
+  name                = "acctest-vscg-%s"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = "%s"
+  connectivity        = "PublicAddress"
+  e911_type           = "Standard"
+  codecs              = "PCMA"
+  platforms           = ["OperatorConnect"]
+  on_prem_mcp_enabled = false
+
+  service_locations {
+    location           = "eastus"
+    operator_addresses = ["198.51.100.1"]
+  }
+
+  service_locations {
+    location           = "eastus2"
+    operator_addresses = ["198.51.100.2"]
+  }
+}
+`, template, data.RandomString, data.Locations.Primary)
+}
+
+func (r VoiceServicesCommunicationsGatewayTestResource) requiresImport(data acceptance.TestData) string {
+	config := r.basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_voice_services_communications_gateway" "import" {
+  name                = azurerm_voice_services_communications_gateway.test.name
+  resource_group_name = azurerm_voice_services_communications_gateway.test.resource_group_name
+  location            = azurerm_voice_services_communications_gateway.test.location
+  connectivity        = azurerm_voice_services_communications_gateway.test.connectivity
+  e911_type           = azurerm_voice_services_communications_gateway.test.e911_type
+  codecs              = azurerm_voice_services_communications_gateway.test.codecs
+  platforms           = azurerm_voice_services_communications_gateway.test.platforms
+  on_prem_mcp_enabled = azurerm_voice_services_communications_gateway.test.on_prem_mcp_enabled
+
+  service_locations {
+    location                                  = "useast"
+    allowed_media_source_address_prefixes     = ["10.1.2.0/24"]
+    allowed_signaling_source_address_prefixes = ["10.1.1.0/24"]
+    operator_addresses                        = ["198.51.100.1"]
+  }
+
+  service_locations {
+    location                                  = "eastus2"
+    allowed_media_source_address_prefixes     = ["10.2.2.0/24"]
+    allowed_signaling_source_address_prefixes = ["10.2.1.0/24"]
+    operator_addresses                        = ["198.51.100.2"]
+  }
+}
+`, config)
+}
+
+func (r VoiceServicesCommunicationsGatewayTestResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+
+%s
+
+resource "azurerm_voice_services_communications_gateway" "test" {
+  name                = "acctest-vscg-%s"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = "%s"
+
+  connectivity = "PublicAddress"
+  e911_type    = "DirectToEsrp"
+  codecs       = "PCMA"
+  platforms    = ["OperatorConnect", "TeamsPhoneMobile"]
+
+  service_locations {
```

Comment:
this should be 
```suggestion
  service_location {
```
as its a block?

Reply:
@katbyte thanks for the feedback. The code has been updated. Could you please take another look?